mypy/semanal.py

```diff
@@ -3969,6 +3969,12 @@ def lookup_qualified(self, name: str, ctx: Context,
                     namespace = node.fullname
                 elif isinstance(node, PlaceholderNode):
                     return sym
+                elif (isinstance(node, TypeAlias) and
+                      isinstance(get_proper_type(node.target), Instance)):
```

Comment:
I think this should be supported only for `no_args` aliases, i.e. you can almost literally copy the code from `visit_member_expr()`. Btw, it seems to me the TODO there is no more relevant, you can also try adding a test for situation like `A = Out; B = A; x: B.In`.